### PR TITLE
Update to enhance Cytomining Ecosystem findings

### PR DESCRIPTION
**Pull request by d33bs**
This PR adds various content and additional findings to the Cytomining Ecosystem analysis. Thanks in advance for your feedback on this!

### CODE REVIEW

File: .codespell-ignore [hunk 1]
```diff
@@ -0,0 +1 @@
+theis
```
Comment (gwaybio) [inline .codespell-ignore]: Can you add a comment describing why this is necessary?

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 2]
```diff
@@ -0,0 +1,688 @@
...
+        "plot": fig_usage_stars,
+        "description": """
+        This plot explores the user bases of relevant projects
+        through relative GitHub stars and the age of the project in years.
+        GitHub stars is an indication of unique users whom are interested
+        in and likely use the project somehow.
+        """,
+        "findings": """
+        We observe that projects of greater age are not always those with the most stars.
+        Projects of <code>loi-focus</code> category are generally
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        Projects of the <code>loi-focus</code> category  (e.g., pycytominer) are generally
```

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 3]
```diff
@@ -0,0 +1,688 @@
...
+        in an early state of their user base (both in age and stars),
+        relative to other projects.
+        """,
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        relative to other projects (e.g., Napari). We note the limitations of using stars to gauge community engagement given many projects do not focus efforts on developers.
        """,
```

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 4]
```diff
@@ -0,0 +1,688 @@
...
+
+fig_collection.append(
+    {
+        "plot": fig_usage_stars_treemap,
+        "description": """
+        This plot demonstrates the magnitude of total GitHub stars by project and category.
+        Click the category boxes to zoom into focus on only those projects.
+        """,
+        "findings": """
+        Here we can see relevant open source far outsize other categories.
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        Here we can see the <code>relevant open source</code> category far outsize other categories.
```

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 5]
```diff
@@ -0,0 +1,688 @@
...
+        Within the microscopy analysis tools category we can observe that Napari has many more stars than others.
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        Within the <code>microscopy analysis tools</code> category we observe that Napari has many more stars than others.
```

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 6]
```diff
@@ -0,0 +1,688 @@
...
+        Projects of <code>loi-focus</code> are nearly invisible in comparison to other projects with
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        Projects of the <code>loi-focus</code> category are emerging with a growing community base, but not are not as established as other tools with
```

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 7]
```diff
@@ -0,0 +1,688 @@
...
+    yaxis_title="Total Lines of Code (Log Scale)",
+    xaxis_title="Project Age (years)",
+)
+
+fig_collection.append(
+    {
+        "plot": fig_maturity_loc_and_age,
+        "description": """
+        This plot shows the lines of code (LoC) and age in years to help indicate relative project maturity.
+        Lines of code in this case are the sum total of GitHub detected programming language dialects.
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        Lines of code in this case are the sum total of all detected programming languages in the latest GitHub repository.
```

Accurate? I tried to simplify/clarify

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 8]
```diff
@@ -0,0 +1,688 @@
...
+        """,
+        "findings": """
+        We can observe that many tools within the Cytomining Ecosystem landscape reside within a certain range
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        We observe that many tools within the Cytomining Ecosystem landscape reside within a tight range
```

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 9]
```diff
@@ -0,0 +1,688 @@
...
+        of total LoC and that projects of greater age do not necessarily have more code.
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        of total lines of code and that projects of greater age do not necessarily have more code.
```

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py (hunk 8 above)
Comment (gwaybio) [reply notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]: please correct - I haven't seen the plot.

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 10]
```diff
@@ -0,0 +1,688 @@
...
+        Tools of <code>loi-focus</code> are nearby the LoC of other tools with greater age.
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
```

I think this is redundant with the first sentence, but please clarify if not

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 11]
```diff
@@ -0,0 +1,688 @@
...
+    {
+        "plot": fig_maturity_latest_commit,
+        "description": """
+        This plot demonstrates project GitHub stars and most recent commit datetime as a way to observe
+        potential decay for projects which may have been popular at one point but are no longer maintained.
+        """,
+        "findings": """
+        We find that relevant open source, microscopy analysis tools, and tools of <code>loi-focus</code> are
+        all recently updated. Adjacent tools demonstrate historical usage but are somewhat less frequently maintained.
+        Many tools from GitHub queries show a lack of updates for greater than 2 years.
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py:290]:
```suggestion
        Many tools from GitHub queries show a lack of updates for greater than 2 years.
        This indicates that many GitHub projects that may re-invent Cytomining functionality are likely abandoned after the authors perform bespoke analyses.
```

True? Or too much of a stretch?

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 12]
```diff
@@ -0,0 +1,688 @@
...
+
+fig_collection.append(
+    {
+        "plot": fig_network_and_subscribers,
+        "description": """
+        This plot shows project GitHub watcher and GitHub fork count
+        as a way to demonstrate relative usage.
+        """,
+        "findings": """
+        We can see a wide variety of project watcher and fork count depending on the category involved.
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        We see a wide variety of project watcher and fork count depending on the category.
```

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 13]
```diff
@@ -0,0 +1,688 @@
...
+        Generally, projects of <code>loi-focus</code> are at an earlier stage and thus have fewer forks and watchers.
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        Generally, projects of <code>loi-focus</code> are emerging tools that are at an earlier stage and thus have fewer forks and watchers.
```

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 14]
```diff
@@ -0,0 +1,688 @@
...
+fig_collection.append(
+    {
+        "plot": fig_contributors_and_issues,
+        "description": """
+        This plot shows the number of project contributors and issue counts as a way to describe usage through maintainability.
+        """,
+        "findings": """
+        We find that many projects have a high contributor and issue count.
+        A few Cytomining Ecosystem adjacent and microscopy analysis tools appear to be on a trend towards
+        relevant open source tools.
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        A few examples in the <code>Cytomining Ecosystem adjacent<code> and <code>microscopy analysis tool<code> categories are trending toward tools in the <code>relevant open source<code> category.
        relevant open source tools.
```

Comment (gwaybio) [reply notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]: I may have gotten one of the category names wrong

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 15]
```diff
@@ -0,0 +1,688 @@
...
+
+fig_collection.append(
+    {
+        "plot": fig_orgs,
+        "description": """
+        This plot shows the count of projects per GitHub organization (a collection of repositories)
+        for those organizations with more than 2 projects.
+        """,
+        "findings": """
+        Here we find that the Theis Lab, UC Davis Bioinformatics Training, and Teich Lab are at the top of this listing.
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]:
```suggestion
        Here we find that the Theis Lab, UC Davis Bioinformatics Training, and Teich Lab are at the top of this listing.
        Cytomining is also near the top of the list, which is a code base that supports essential functionality that does not exist in any other open source organization that is actively maintained.
```

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py [hunk 16]
```diff
@@ -0,0 +1,688 @@
...
+)
+
+fig_collection.append(
+    {
+        "plot": fig_orgs_stars,
+        "description": """
+        This plot shows the sum of GitHub stars per GitHub organization for those with a sum total of stars greater than 100.
+        """,
+        "findings": """
+        Here we can observe that scverse, Theis Lab, and Satija Lab organizations hold the top count of GitHub stars.
```
Comment (gwaybio) [inline notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]: Is there one sentence to add about Cytomining? It is nice to keep this focus in the findings section

**Review by gwaybio — APPROVED** [review]
Ugh, I literally just realized I made suggestions in the nbconverted file. Sorry!! I hope it's not too much trouble to transfer over to the notebook tomorrow.

Please merge this once you have the time and you're happy! It is looking great!

-------
EDIT: One other note: Please make sure to add a link to this repo https://github.com/WayScience/software-landscape-analysis somewhere in the landing page of the io site. Thanks! (if it isn't already there)

File: .codespell-ignore (hunk 1 above)
Comment (d33bs) [reply .codespell-ignore]: Thanks - this gets flagged by codespell as related to "this" or "thesis" as part of the findings in the visualization notebook. I'll add a comment here to help describe this.

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py (hunk 2 above)
Comment (d33bs) [reply notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]: Thanks - I'll add this in!

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py (hunk 7 above)
Comment (d33bs) [reply notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]: Thanks - yes this is accurate, I'll add this in!

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py (hunk 8 above)
Comment (d33bs) [reply notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]: Thanks - yes this is accurate, adding this in.

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py (hunk 10 above)
Comment (d33bs) [reply notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]: Thanks, agreed, removing.

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py (hunk 11 above)
Comment (d33bs) [reply notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py:290]: Thanks - I think this is mostly true, but I softened the suggestions here as hypotheticals (I don't know if we have enough evidence yet solely from GitHub metrics to state as a fact).

Adjusted:
```text
Many tools from GitHub queries show a lack of updates for greater than 2 years.
We hypothesize that many GitHub projects re-invent functionality related to the Cytomining Ecosystem.
These projects are sometimes abandoned after authors perform bespoke or one-time analyses.
```

Food for thought: I think in analyzing the readme's of these projects we could ascertain the "bespokeness" of individual projects (i.e. text like "this repo relates to <single dataset> and <single methods> and <single publication>"). I imagine but don't know that we could end up with NLP-defined classes, and that one of those classes would likely be "one-time research projects" (or something of that nature).

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py (hunk 14 above)
Comment (d33bs) [reply notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]: Thanks - adding this in!

File: notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py (hunk 16 above)
Comment (d33bs) [reply notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py]: Thanks, added a statement: `The Cytomining organization is found near the middle of this visualization, indicating similar popularity and early success in audience outreach amidst those within the related landscape.`

File: .codespell-ignore (hunk 1 above)
Comment (d33bs) [reply .codespell-ignore]: I moved to an inline-based ignore system instead for this to avoid adding another file.

**Comment by d33bs** [comment]
Thanks @gwaybio, merging this in after changes!

**Pull request merged**